rest_framework/compat.py

```diff
@@ -152,6 +152,25 @@ def md_filter_add_syntax_highlight(md):
         return False
 
 
+try:
+    # Django <= 4.1, create a compatibility wrapper for parse_header_parameters
+    from django.http.multipartparser import parse_header
+
+    def parse_header_parameters(line):
+        main_value_pair, params = parse_header(line.encode())
+        return main_value_pair, {
+            k: v if isinstance(v, str) else v.decode()
+            for k, v in params.items()
+        }
+
+except ImportError:
+    # Django 4.2+, use the new built-in function
+    # Note: Django 4.1 also has an implementation of parse_header_parameters
+    #       which is slightly different from the one in 4.2
+    from django.utils.http import parse_header_parameters
+    parse_header_parameters = parse_header_parameters
+
+
```

Comment:
Normally we'd import the future version, falling back to the shim, so... 

```python
try: 
    from django.utils.http import parse_header_parameters
except ImportError:
   from django.http.multipartparser import parse_header
 
   def parse_header_parameters(...):
       ...

Reply:
Yes, but as I noted in the comment:

# Note: Django 4.1 also has an implementation of parse_header_parameters
#       which is slightly different from the one in 4.2

Reply:
I could do an explicit Django version check here, (`if django.VERSION > (4, 1):`) if you'd prefer that

Reply:
I'm used to reading the new version first, so yes I think I do prefer it that way round. Thanks!